### PR TITLE
storage: allow to use a different configuration file

### PR DESCRIPTION
**Pull request by giuseppe**
Move the configuration file parsing to a new function `ReloadConfigurationFile` so that it is possible to load a differentconfiguration file and override the current settings.  This is useful with rootless containers so a configuration file specific to the user can be loaded.

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -3004,11 +3004,7 @@ type tomlConfig struct {
 	} `toml:"storage"`
 }
 
-func init() {
-	DefaultStoreOptions.RunRoot = "/var/run/containers/storage"
-	DefaultStoreOptions.GraphRoot = "/var/lib/containers/storage"
-	DefaultStoreOptions.GraphDriverName = ""
-
+func ReloadConfigurationFile(configFile string, storeOptions *StoreOptions) {
```
Comment (rhatdan) [inline store.go]: Add a comment defining this function.

File: storage.conf [hunk 2]
```diff
@@ -25,6 +25,11 @@ additionalimagestores = [
 # certain container storage drivers.
 size = ""
 
+# Path to a FUSE helper to use for mounting the file system instead of mounting it
+# directly.
+#fuse_program = "/usr/bin/fuse-overlayfs"
```
Comment (rhatdan) [inline storage.conf]: Should we call this mount_program rather then fuse.  I see no reason to only do this for fuse executables.
```
#mount_program = "/usr/bin/fuse-overlayfs"
```

File: store.go [hunk 3]
```diff
@@ -2992,6 +2992,9 @@ type OptionsConfig struct {
 
 	// Do not create a bind mount on the storage home
 	SkipMountHome string `toml:"skip_mount_home"`
+
+	// Alternative program to use for the mount of the file system
+	FuseProgram string `toml:"fuse_program"`
```
Comment (rhatdan) [inline store.go]: MountProgram ...

File: store.go (hunk 1 above)
Comment (giuseppe) [reply store.go]: done

File: storage.conf (hunk 2 above)
Comment (giuseppe) [reply storage.conf]: this was already present in containers/storage.  So I've added a new commit to rename the option name everywhere

File: store.go (hunk 3 above)
Comment (giuseppe) [reply store.go]: done, I've renamed Fuse -> Mount everywhere

**Comment by rhatdan** [comment]
LGTM
@nalind PTAL

File: docs/containers-storage.conf.5.md [hunk 4]
```diff
@@ -89,9 +89,9 @@ Tells driver to wipe device (directlvm_device) even if device already has a file
 
 Specifies the filesystem type to use for the base device. (default: xfs)
 
-**fuse_program**=""
+**mount_program**=""
 
-Specifies the path to a custom FUSE program to use instead for mounting the file system.
+Specifies the path to a custom program to use instead for mounting the file system.
```
Comment (nalind) [inline docs/containers-storage.conf.5.md]: This shouldn't be in the `storage.options.thinpool` section.

File: store.go [hunk 5]
```diff
@@ -3024,72 +3025,78 @@ func init() {
...
 	if config.Storage.Options.SkipMountHome != "" {
-		DefaultStoreOptions.GraphDriverOptions = append(DefaultStoreOptions.GraphDriverOptions, fmt.Sprintf("%s.skip_mount_home=%s", config.Storage.Driver, config.Storage.Options.SkipMountHome))
+		storeOptions.GraphDriverOptions = append(storeOptions.GraphDriverOptions, fmt.Sprintf("%s.skip_mount_home=%s", config.Storage.Driver, config.Storage.Options.SkipMountHome))
+	}
+	if config.Storage.Options.MountProgram != "" {
+		storeOptions.GraphDriverOptions = append(storeOptions.GraphDriverOptions, fmt.Sprintf("%s.mount_program=%s", config.Storage.Driver, config.Storage.Options.MountProgram))
+	}
+	if config.Storage.Options.OverrideKernelCheck != "" {
+		storeOptions.GraphDriverOptions = append(storeOptions.GraphDriverOptions, fmt.Sprintf("%s.override_kernel_check=%s", config.Storage.Driver, config.Storage.Options.OverrideKernelCheck))
 	}
```
Comment (nalind) [inline store.go]: This block duplicates the block that follows it.

**Comment by nalind** [comment]
A couple of comments, but mostly fine.

File: docs/containers-storage.conf.5.md (hunk 4 above)
Comment (giuseppe) [reply docs/containers-storage.conf.5.md]: thanks, I've moved it

File: store.go (hunk 5 above)
Comment (giuseppe) [reply store.go]: dropped

**Pull request merged**